debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractReadOnlyIncrementalSnapshotChangeEventSource.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.pipeline.source.snapshot.incremental;
+
+import java.util.List;
+
+import io.debezium.DebeziumException;
+import io.debezium.annotation.NotThreadSafe;
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.connector.common.RelationalBaseSourceConnector;
+import io.debezium.jdbc.JdbcConnection;
+import io.debezium.pipeline.EventDispatcher;
+import io.debezium.pipeline.signal.AbstractExternalSignalThread;
+import io.debezium.pipeline.signal.ExternalSignalThread;
+import io.debezium.pipeline.source.spi.DataChangeEventListener;
+import io.debezium.pipeline.source.spi.SnapshotProgressListener;
+import io.debezium.relational.RelationalDatabaseConnectorConfig;
+import io.debezium.schema.DataCollectionId;
+import io.debezium.schema.DatabaseSchema;
+import io.debezium.util.Clock;
+
+/**
+ * An read only incremental snapshot change event source that emits events from a DB log interleaved with snapshot events.
+ */
+@NotThreadSafe
+public abstract class AbstractReadOnlyIncrementalSnapshotChangeEventSource<T extends DataCollectionId> extends AbstractIncrementalSnapshotChangeEventSource<T> {
+
+    protected final ExternalSignalThread externalSignal;
+
+    public AbstractReadOnlyIncrementalSnapshotChangeEventSource(RelationalDatabaseConnectorConfig config,
+                                                                JdbcConnection jdbcConnection,
+                                                                EventDispatcher<T> dispatcher, DatabaseSchema<?> databaseSchema,
+                                                                Clock clock, SnapshotProgressListener progressListener,
+                                                                DataChangeEventListener dataChangeEventListener) {
+        super(config, jdbcConnection, dispatcher, databaseSchema, clock, progressListener, dataChangeEventListener);
+
+        // externalSignal = new KafkaSignalThread<>(MySqlConnector.class, config, this);
+        // dynamically load signaling source
+        // @TODO do we need Factory class here?
+        final String externalSignalClassName = config.getConfig().getString("signal.class", "io.debezium.pipeline.signal.KafkaSignalThread");
+        try {
+            // @TODO not sure this is correct way to get connectorType
+            final String connectorClassName = config.getConfig().getString("connector.class");
```

Comment:
i believe we need to get connector class here to instantiate signal thread here?